components/gly_gen/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { MetaNode } from '@/spec/metanode'
+import { GeneInfo } from '../service/mygeneinfo'
+import { z } from 'zod'
+
+export const GlyGenGeneType = {
+  gene_name: typeof GeneInfo
+}
+
+export const GlyGenResponse = z.object({
+  queryinfo: z.any(),
+  results: z.array(z.any())
+})
+
+export type GlyGenResponseType = z.infer<typeof GlyGenResponse>
+
+export const GlyGenResponseNode = MetaNode.createData('GlyGenResponse')
+  .meta({
+    label: 'GlyGen Response',
+    description: 'GlyGen response object',
+  })
+  .codec(GlyGenResponse)
+  .view(data => (
+    <>
+      <div><pre>{JSON.stringify(data)}</pre></div>
+    </>
+    
+  ))
+  .build()
+
+export const ProteinProductInformation = MetaNode.createProcess('ProteinProductInformation')
+  .meta({
+    label: 'Protein Product Information',
+    description: 'Search for protein records in GlyGen',
+  })
+  .inputs({ gene: GeneInfo })
+  .output(GlyGenResponseNode)
+  .resolve(async (props) => {
+    const query = encodeURIComponent(`{"recommended_gene_name":"${props.inputs.gene.symbol}"`)
+    console.log('props', props, query)
+    const response = await fetch(`https://api.glygen.org/directsearch/protein/?query=${query}`, {
+      method: 'GET',
+      headers: {
+        accept: 'application/json'
+      }
+      })
+    return response.json()
```

Comment:
So the way `fetch` works in javascript is in two steps I usually write as follows:

```
const request = await fetch('...')
const response = await request.json()
// your data is now in response
```

Namely you've returned `response.json()` which is a promise, but you need to `await` it to get the actual data.

Reply:
I modify those lines to be:

```
  .resolve(async (props) => {
    const query = encodeURIComponent(`{"recommended_gene_name":"${props.inputs.gene.symbol}"`)
    console.log('props', props, query)
    const request = await fetch(`https://api.glygen.org/directsearch/protein/?query=${query}`, {
      method: 'GET',
      headers: {
        accept: 'application/json'
      }
      })
    const response = await request.json()
    console.log('response', response)
```

And I get in the terminal (from console.log):
```
props {
  inputs: {
    gene: {
      _id: '59272',
      symbol: 'ACE2',
      entrezgene: '59272',
      name: 'angiotensin converting enzyme 2',
      taxid: 9606,
      ensembl: [Object]
    }
  }
} %7B%22recommended_gene_name%22%3A%22ACE2%22
response { error_list: [ { error_code: 'invalid-query-json' } ] }
props {
  inputs: {
    gene: {
      _id: '59272',
      symbol: 'ACE2',
      entrezgene: '59272',
      name: 'angiotensin converting enzyme 2',
      taxid: 9606,
      ensembl: [Object]
    }
  }
} %7B%22recommended_gene_name%22%3A%22ACE2%22
response { error_list: [ { error_code: 'invalid-query-json' } ] }
```

I seem to have something mixed up with the type script object declarations? I am new to type script but not so much react.

Reply:
So the actual API request happens server side so you won't see it in the console unfortunately.

In the code you provided there seems to be an error:

```ts
    const query = encodeURIComponent(`{"recommended_gene_name":"${props.inputs.gene.symbol}"`)
```

Namely you're missing another `}` here: `ol}"`)` => `ol}"}`)`

I recommend:
```ts
const query = encodeURIComponent(JSON.stringify({
  recommended_gene_name: props.inputs.gene.symbol,
}))
```
And let javascript make sure the json is valid